malaria24/ona/tests/test_tasks.py

```diff
@@ -174,6 +175,177 @@ def test_compile_and_send_digest_email(self):
             mis.email_address]))
         self.assertEqual(len(mail.outbox), 4)
 
+    @responses.activate
+    def test_national_digest_week_variable(self):
+        self.mk_actor(role=MIS,
+                      email_address='manager@example.org')
+        Facility.objects.create(facility_code='342315',
+                                facility_name='Facility 1',
+                                province='Limpopo',
+                                district=u'Example1')
+        self.mk_actor(role=MANAGER_DISTRICT,
+                      email_address='manager@example.org',
+                      facility_code='342315')
+        self.mk_actor(role=MANAGER_PROVINCIAL,
+                      email_address='m2@example.org',
+                      facility_code='342315')
+        self.mk_actor(role=MANAGER_NATIONAL,
+                      email_address='m3@example.org',
+                      facility_code='342315')
+        ehp1 = self.mk_ehp(name='EHP1', email_address='ehp1@example.org')
+        ehp2 = self.mk_ehp(name='EHP2', email_address='ehp2@example.org')
+
+        for i in range(10):
+            case = self.mk_case(gender='female', facility_code='342315')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+
+        for i in range(10):
+            case = self.mk_case(gender='male', facility_code='222222')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+
+        for i in range(10):
+            case = self.mk_case(gender='male', facility_code='333333')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+        self.mk_case()
+        compile_and_send_digest_email()
+        NationalDigest.compile_digest().send_digest_email()
```

Comment:
I think we can leave this line out, since we don't really want to test that we're sending the email (and we're not doing any asserts on this anyway), we just want to test that the correct data is being sent to the send email function. So doing the `compile_digest()` and `get_digest_email_data()` below, and then asserting on those results, will be enough to test this.

Same for the other tests.

Reply:
Won't this skip over the error found in QA? because it was triggered when the code reached send_digest_email(), which in turn called the line context = self.get_digest_email_data(), where then the error was in

Reply:
Below in line 225, we're calling `get_digest_email_data` ourselves anyway. So whether we call `compile_and_send_digest_email`, and that calls `get_digest_email_data`, or whether we call `send_digest_mail`, and that calls `get_digest_email_data`, or if we call `get_digest_email_data` directly, all of them will result in a call to `get_digest_email_data`, where the error is. So it's unnecessary to call `get_digest_email_data` in a variety of ways, we just need to call it one way.

Reply:
Noted, and agreed. Thanks for pointing it out